benchopt/plotting/generate_html.py

```diff
@@ -0,0 +1,296 @@
+
+    return results
+
+
+def get_results(fnames, kinds, copy=False):
+    results = []
+    BUILD_DIR_OUTPUTS.mkdir(exist_ok=True, parents=True)
+    BUILD_DIR_FIGURES.mkdir(exist_ok=True, parents=True)
+
+    for fname in fnames:
+        print(f"Processing {fname}")
+
+        df = pd.read_csv(fname)
+        datasets = list(df['data_name'].unique())
+
+        # Copy CSV
+        if copy:
+            fname_no_output = fname.replace('outputs/', '')
+            fname_no_output_path = Path(fname_no_output)
+            shutil.copy(fname, BUILD_DIR_OUTPUTS / fname_no_output_path)
+
+        # Generate figures
+        result = dict(
+            fname=fname,
+            datasets=datasets,
+            **generate_plot_benchmark(df, kinds, fname)
+        )
+        results.append(result)
+
+    for result in results:
+        result['page'] = \
+            f"{str(result['fname_short']).replace('.csv', '.html')}"
+
+    return results
+
+
+def render_benchmark(results, benchmark):
+    return Template(filename=str(TEMPLATE_BENCHMARK),
+                    input_encoding="utf-8").render(
+        results=results,
+        benchmark=benchmark,
+        max_rows=15,
+        nb_total_benchs=len(results),
+        last_updated=datetime.now(),
+    )
+
+
+def render_index(benchmarks):
+    return Template(filename=str(TEMPLATE_INDEX),
+                    input_encoding="utf-8").render(
+        benchmarks=benchmarks,
+        nb_total_benchs=len(benchmarks),
+        max_rows=15,
+        last_updated=datetime.now(),
+    )
+
+
+def copy_static(benchmark=None):
+    if benchmark is not None:
+        dst = benchmark / BUILD_DIR / 'static'
+    else:
+        dst = BUILD_DIR / 'static'
+    if dst.exists():
+        shutil.rmtree(dst)
+    shutil.copytree(ROOT / 'static', dst)
+
+
+def render_all_results(results, benchmark, local=True):
+    if local:
+        template = TEMPLATE_LOCAL_RESULT
+    else:
+        template = TEMPLATE_RESULT
+    htmls = []
+    for result in results:
+        html = Template(
+            filename=str(template),
+            input_encoding="utf-8").render(
+            result=result,
+            benchmark=benchmark,
+        )
+        htmls.append(html)
+    return htmls
+
+
+def plot_benchmark_html(df, output_dir, kinds,
+                        display=True, index=False):
+    """
+    Plot convergence curves and histograms for a
+    given benchmark in an HTML file using plotly.
+
+    Parameters
+    ----------
+    df : instance of pandas.DataFrame
+        The benchmark results.
+    benchmark_instance : Benchmark
+        The benchmark instance.
+    display : bool
+        If set to True, display the curves by opening
+        the default browser.
+
+    Returns
+    -------
+    None
+    """
+
+    benchmark = output_dir
+    try:
+        print("Removing old html files")
+        shutil.rmtree(benchmark / BUILD_DIR)
+    except FileNotFoundError:
+        print("Building html files")
```

Comment:
```suggestion
    print("Removing old html files")
    if (benchmark / BUILD_DIR).exists():
        shutil.rmtree(benchmark / BUILD_DIR)

     print("Building html files")
```

Reply:
Actually, I am not sure you want to delete the previous files.

Overall, I would say that when you generate the html, you want:
- to generate the html for the current dataframe.
- update the `benchmark_filename = BUILD_DIR / f"{benchmark}.html"` to  reference this `run`

That way you can aggregate the results and navigate easily all the results you have.

Reply:
When you say navigate, you want another html file to reference all the runs, as a "home page" ?